### PR TITLE
feat(docs): typeahead-use-cases

### PR DESCRIPTION
**Pull request by kulenkov**
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Append to body functionality
+
+Main success scenario:
+----------------------
+1. User opens Typeahead demo page
+2. User clicks on Append to body sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text
+4. When user starts to type a name of a State from "states" array then order of typed symbols and spaces between them is ignored then a drop-down with a list of States matches is shown
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: then a drop-down => and a drop-down

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+5. If there are no matches then the drop-down is hidden. The "Model" is filled with inputted data
+6. When there are any matches then a drop-down with a list of States matches is shown.
+7. Items in the drop-down are clickable.
+8. When user clicks on any item in typeahead drop-down, then typeahead container auto-fills with a selected State
+
+Extensions:
+-----------
+3a. If there is any data, it could be deleted. While deleting data the drop-down with matches is shown
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: this is an extention not for 3

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+16.13: Typeahead Append to body example
+=======================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Append to body functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.async-data.use-case.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+16.4: Typeahead Async data example
+==================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Async data functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.async-data.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.grouping-results.use-case.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+16.8: Typeahead Grouping results  example
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.grouping-results.use-case.md]: results  example - 2 spaces

File: docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+16.9: Typeahead Ignore spaces and order example
+==============================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Ignore spaces and symbols order functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+Main success scenario:
+----------------------
+1. User opens Typeahead demo page
+2. User clicks on Ignore spaces and order sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text. User sees the button "Toggle typeaheadSingleWords"
+`Default:` the input property is set `typeaheadSingleWords = true`. Order of typed symbols and spaces between them will is ignored
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md]: it should be 4th. By default, ...
them will is ?

File: docs/spec/typeahead/typeahead.examples.item-template.use-case.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+16.2: Typeahead Item template example
+=====================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Item template functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.item-template.use-case.md:7]: rewrite

File: docs/spec/typeahead/typeahead.examples.item-template.use-case.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+Main success scenario:
+----------------------
+1. User opens Typeahead demo page
+2. User click on Item template sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text
+4. When user starts to type a name of a State from "states" array and there is no matches the drop-down is hidden. The "Model" is filled with inputted data
+5. When there are any matches then a drop-down with a list of States matches is shown. The list contains a template "This is: "_matched state_" index: _index #"_
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.item-template.use-case.md:15]: there are => there is

File: docs/spec/typeahead/typeahead.examples.latinize.use-case.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+16.16: Typeahead Latinize example
+=================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Latinize functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.latinize.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
+16.14: Typeahead No results example
+===================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user No results functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+Main success scenario:
+----------------------
+1. User opens Typeahead demo page
+2. User clicks on No results sub-menu
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: No result

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
+16.14: Typeahead No results example
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: No result
and rewrite others

File: docs/spec/typeahead/typeahead.examples.on-blur.use-case.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
+16.12: Typeahead On blur example
+================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user On blur functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.on-blur.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.on-blur.use-case.md [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+Main success scenario:
+----------------------
+1. User opens Typeahead demo page
+2. User clicks on On blur sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text. User sees typeahead card with "Option on blur:" text
+4. When user starts to type a name of a State from "states" array and there is no matches the drop-down is hidden
+5. If user clicks anywhere outside the input than the "Option on blur" stays empty
+6. When there are any matches then a drop-down with a list of States matches is shown
+7. If user clicks anywhere outside the input than the "Option on blur" is filled with the first item from the States matches
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.on-blur.use-case.md]: If user clicks anywhere outside the input than the "Option on blur" is filled with the first item from the States matches - need rewrite.
logic is another.

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md [hunk 16]
```diff
@@ -0,0 +1,25 @@
+16.17: Typeahead On select example
+==================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user On select functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Selected sub-menu
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: there is no such menu

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+6. Items in the drop-down are clickable
+7. When user clicks on any item in typeahead drop-down, then typeahead container auto-fills with a selected State and "Selected option" is filled with info about selected object: "id", "name" and "region"
+
+Extensions:
+-----------
+3a. If there is any data, it could be deleted. While deleting data the drop-down with matches is shown
+
+Variations:
+-----------
+2*. User scrolls to On selected sub-menu
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: there is no such menu

File: docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md [hunk 19]
```diff
@@ -0,0 +1,26 @@
+16.10: Typeahead Phrase delimiters example
+====================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Phrase delimiters functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Phrase delimiters sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text
+4. User is able to use "&" and "," to delimit words
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md]: question to PO

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
+16.7: Typeahead Reactive forms example
+=============================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Reactive forms functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on Reactive forms forms sub-menu
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: forms forms

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+3. User sees typeahead input and typeahead card with "Model:" text and it's filled with "null". The input has a placeholder "Typeahead inside a form"
+4. When user places mouse coursor in the "Model" then a drop-down with the first seven States is shown
+5. When user starts to type a name of a State from "states" array and there are no matches then then the drop-down is hidden. The "Model" is filled with inputted data
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: then then

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+7. Items in the drop-down are clickable
+8. When user clicks on any item in typeahead drop-down, then typeahead container auto-fills with a selected State
+
+Extensions:
+-----------
+3a. "Model:" is already filled with "null"
+
+Variations:
+-----------
+2*. User scrolls to Reactive forms forms sub-menu
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: forms forms

File: docs/spec/typeahead/typeahead.examples.scrollable.use-case.md [hunk 25]
```diff
@@ -0,0 +1,26 @@
+16.15: Typeahead Scrollable example
+===================================
+**Primary Actor**: User
+
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user Scrollable drop-down functionality
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.scrollable.use-case.md]: rewrite

File: docs/spec/typeahead/typeahead.examples.with-delay.use-case.md [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+**Scope**: Ngx-bootstrap DEMO / BS version 3&4
+
+**Goal**: Show user With delay functionality
+
+Main success scenario:
+----------------------
+1. User open Typeahead demo page
+2. User click on With delay sub-menu
+3. User sees typeahead input and typeahead card with "Model:" text
+4. When user starts to type a name of a State from "states" array Drop-dawn update delay depends on key `typeaheadWaitMs`, time delay = 1 sec, by default
```
Comment (ludmilanesvitiy) [inline docs/spec/typeahead/typeahead.examples.with-delay.use-case.md]: Drop-down

**Review by ludmilanesvitiy — CHANGES_REQUESTED** [review]

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md (hunk 1 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.grouping-results.use-case.md (hunk 5 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.grouping-results.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md (hunk 13 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md (hunk 12 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.on-blur.use-case.md (hunk 15 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.on-blur.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md (hunk 17 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md (hunk 18 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md (hunk 21 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md (hunk 22 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md (hunk 23 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md (hunk 24 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.with-delay.use-case.md (hunk 26 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.with-delay.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md (hunk 3 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md (hunk 2 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.append-to-body.use-case.md]: deleted here and in the other use-cases

File: docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md (hunk 6 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md (hunk 7 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.ignore-spaces-order.use-case.md]: fixed

File: docs/spec/typeahead/typeahead.examples.item-template.use-case.md (hunk 8 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.item-template.use-case.md:7]: re-written

File: docs/spec/typeahead/typeahead.examples.item-template.use-case.md (hunk 9 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.item-template.use-case.md:15]: i'm not sure about these. matches = plural

File: docs/spec/typeahead/typeahead.examples.latinize.use-case.md (hunk 10 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.latinize.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.no-results.use-case.md (hunk 11 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.no-results.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.on-blur.use-case.md (hunk 14 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.on-blur.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.scrollable.use-case.md (hunk 25 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.scrollable.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.on-select.use-case.md (hunk 16 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.on-select.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md (hunk 19 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md (hunk 21 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.reactive-forms.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.scrollable.use-case.md (hunk 25 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.scrollable.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.async-data.use-case.md (hunk 4 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.async-data.use-case.md]: re-written

File: docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md (hunk 20 above)
Comment (kulenkov) [reply docs/spec/typeahead/typeahead.examples.phrase-delimiters.use-case.md]: done. steps re-written

**Review by ludmilanesvitiy — APPROVED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**